api/src/services/Datastream.ts

```diff
@@ -0,0 +1,39 @@
+import { FedoraObject } from "../models/FedoraObject";
+import FedoraCatalog from "./FedoraCatalog";
+
+class Datastream {
+    private static instance: Datastream;
+    fedoraCatalog: FedoraCatalog;
+
+    constructor(fedoraCatalog: FedoraCatalog) {
+        this.fedoraCatalog = fedoraCatalog;
+    }
+
+    static getInstance(): Datastream {
+        if (!Datastream.instance) {
+            Datastream.instance = new Datastream(FedoraCatalog.getInstance());
+        }
+        return Datastream.instance;
+    }
+
+    async uploadFile(pid: string, stream: string, filepath: string, mimeType: string): Promise<void> {
+        const fedoraObject = FedoraObject.build(pid);
+        if (this.hasValidMimeType(stream, mimeType)) {
+            await fedoraObject.updateDatastreamFromFile(filepath, stream, mimeType);
+        } else {
+            throw new Error("Invalid mime types");
```

Comment:
Would it be worth saying `"Invalid mime type: " + mimeType` instead?

Reply:
Made change